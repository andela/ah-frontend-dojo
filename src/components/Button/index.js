// React libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const Button = (props) => {
  // Destructure props
  const {
    btnClass, btnName, btnEvent, disabled,
  } = props;

  return (

    <button
      className={btnClass}
      type="button"
      onClick={btnEvent}
      disabled={disabled}
    >
      {btnName}
    </button>
  );
};

Button.defaultProps = {
  disabled: false,
};

// Props validation
Button.propTypes = {
  btnName: PropTypes.string.isRequired,
  btnClass: PropTypes.string.isRequired,
  btnEvent: PropTypes.func.isRequired,
  disabled: PropTypes.bool,

};
export default Button;
