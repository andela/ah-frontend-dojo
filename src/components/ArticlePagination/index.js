// react
import React from 'react';

// Proptypes for validation
import PropTypes from 'prop-types';

// stylesheets
import './ArticlePagination.scss';

// Button component
import Button from 'components/Button';

const ArticlePagination = ({
  previous, next, paginate, count, page
}) => {
  const pages = Math.ceil(count / 10);
  return (

    <div className="article-pagination">
      <Button
        btnName=""
        btnClass="fa fa-backward"
        btnEvent={(e) => {
          e.preventDefault();
          paginate(previous, -1);
        }}
        disabled={!previous}
      />
      <span>
        {page || 1}
        {' '}
        {'/'}
        {' '}
        {pages}
        {' '}
        {'pages '}
      </span>
      <Button
        btnName=""
        btnClass="fa fa-forward"
        btnEvent={(e) => {
          e.preventDefault();
          paginate(next, 1);
        }}
        disabled={!next}
      />
    </div>
  );
};

ArticlePagination.defaultProps = {
  next: null,
  previous: null,
};

ArticlePagination.propTypes = {
  next: PropTypes.string,
  page: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
  previous: PropTypes.string,
  paginate: PropTypes.func.isRequired,
};


export default ArticlePagination;
