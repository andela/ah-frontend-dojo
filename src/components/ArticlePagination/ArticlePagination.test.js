import React from 'react';

import { mount } from 'enzyme';

import ArticlePagination from 'components/ArticlePagination';
import Button from 'components/Button';

describe('ArticlePagination', () => {
  const props = {
    previous: null,
    next: 'next',
    paginate: jest.fn(),
    count: 20,
  };
  let wrapper = mount(<ArticlePagination {...props} />);

  it('should render without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  const previousButton = wrapper.find(Button).at(0);
  const nextButton = wrapper.find(Button).at(1);

  it('should disable buttons if null', () => {
    expect(previousButton.props().disabled).toBe(true);
    expect(nextButton.props().disabled).toBe(false);
  });


  it('should call the paginate function on button click', () => {
    const event = {
      target: {
        type: 'button',
        name: '',
      },
      preventDefault: jest.fn(),
    };

    nextButton.simulate('click', event);
    expect(props.paginate).toBeCalled();
    previousButton.simulate('click', event);


    props.previous = 'previous';
    wrapper = mount(<ArticlePagination {...props} />);
    wrapper.find(Button).at(0).simulate('click', event);
    expect(props.paginate).toBeCalled();
  });
});
