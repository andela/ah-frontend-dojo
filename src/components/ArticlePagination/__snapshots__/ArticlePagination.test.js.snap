// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePagination should render without crushing 1`] = `
<ArticlePagination
  count={20}
  next="next"
  paginate={[MockFunction]}
  previous={null}
>
  <div
    className="article-pagination"
  >
    <Button
      btnClass="fa fa-backward"
      btnEvent={[Function]}
      btnName=""
      disabled={true}
    >
      <button
        className="fa fa-backward"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
    </Button>
    <span>
      1
       
      /
       
      2
       
      pages 
    </span>
    <Button
      btnClass="fa fa-forward"
      btnEvent={[Function]}
      btnName=""
      disabled={false}
    >
      <button
        className="fa fa-forward"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </Button>
  </div>
</ArticlePagination>
`;
